docs/src/references/mqtt-topics.md

```diff
@@ -4,15 +4,411 @@ tags: [Reference, MQTT]
+  "temperature": 23.4
+}'
+```
+
+### Telemetry type metadata
+
+The data types also may have additional metadata associated with it,
+which can be added/updated by publishing to `/meta` subtopics of those data types.
+For example, the units associated with measurements in the `battery_reading` measurement type
+can be updated by publishing the following message:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///m/battery_reading/meta '{
+  "units": {
+    "temperature": "C",
+    "voltage": "V",
+    "current": "A"
+  }
+}'
+```
+
+The metadata fields supported by each data type will be defined in detail later.
+
+## Commands
+
+The topic scheme for commands is as follows:
+
+```text
+<entity_id_prefix>/cmd/<cmd-type>/<cmd-id>
+```
+
+|Type|Example|
+|------|-------|
+|Software List|`<entity_id_prefix>/cmd/software_list/<cmd-id>`|
+|Software Update|`<entity_id_prefix>/cmd/software_update/<cmd-id>`|
+|Configuration Snapshot|`<entity_id_prefix>/cmd/config_snapshot/<cmd-id>`|
+|Configuration Update|`<entity_id_prefix>/cmd/config_update/<cmd-id>`|
+|Firmware Update|`<entity_id_prefix>/cmd/firmware_update/<cmd-id>`|
+|Restart|`<entity_id_prefix>/cmd/restart/<cmd-id>`|
+
+The command would be interpreted differently based on the target entity.
+For example, the `restart`` could mean either a device restart or a service restart based on the target entity.
+
+### Examples: With default device/service topic semantics
+
+#### Command to main device
+
+Command to fetch the software list from the main device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/cmd/software_list/123' '{
+  "status": "init"
+}'
+```
+
+The `status` value will transition from `init` to the final `successful` or `failed` via many intermediate states
+such as `validating`, `downloading`, `executing` etc.
+
+#### Command to child device
+
+Command to update the firmware of a child device:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01///cmd/firmware_update/123' '{
+  "status": "init",
+  "attempt": 1,
+  "name": "OpenWRT",
+  "version": "22.03",
+  "url":"http://127.0.0.1:8000/tedge/file-transfer/tedge-child/firmware_update/93d50a297a8c235",
+  "sha256":"c036cbb7553a909f8b8877d4461924307f27ecb66cff928eeeafd569c3887e29"
+}'
+```
+
+#### Command to a service
+
+Command to update the configuration of a service:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main/service/collectd/cmd/config_update/123' '{
+  "status": "init",
+  "type": "collectd"
+  "path": "/etc/collectd/collectd.conf",
+  "url": "http://127.0.0.1:8000/tedge/file-transfer/collectd/config_update/collectd",
+}'
+```
+
+### Commands metadata
+
+For commands as well, additional commad specific metadata can be registered as retained messages on the `<cmd-type>` topic.
+
+For example, the supported configuration list of the main device can be delared as follows:
+
+```sh te2mqtt
+tedge mqtt pub -r te/device/main///cmd/config_snapshot '{
+  "supported-configs": ["mosquitto", "tedge", "collectd"]
```

Comment:
I think we should use camelCase in our JSON fields?

In my other work, I was wondering what is JSON naming convention, and reached to https://stackoverflow.com/questions/5543490/json-naming-convention-snake-case-camelcase-or-pascalcase, which suggests to use camelCase.
Also in c8y and azure, I see they both use camelCase for JSON.

Reply:
yes, I would also vote for camel case in the json payloads.

Reply:
Changed from `supported-configs` to `supportedTypes` as it is already in the configuration context, to adding "config" to the property names is redundant.